src/PrestaShopBundle/Entity/FeatureFlag.php

```diff
@@ -130,6 +130,14 @@ public function getState(): bool
         return $this->state;
     }
 
+    /**
```

Comment:
Please remove getState. It's a duplicate.

Reply:
As I explained in the comment above, I think both `getState()` and `isEnabled()` are valid. I think it's valuable to keep both.

`isEnabled()` tells you if the flag is enabled. It is an accurate need, you want to know if it's enabled, no matter the current value of the object.

`getState()` tells you if the flag state. You use this function to know what is the value of this object state, you do not care what is the value, you want to extract it.

Both functions make sense, both can be used.

Consider the following example
```
class A
{
    public function __toString()
    {
        return $this->name();
    }
    
    public function getName()
    {
        return $this->name();
    }
}
```

These 2 functions look like a duplicate but actually they both make sense and serve different purpose 😄  .

Reply:
Actually, it's a duplicate as you can access at the same information from two places.

Actually, the flag has no state. So when the flag will a state, we will make evolve the code and the structure.

So, could you remove the method `getState()` ?

Reply:
`getState` would be useful if it was more than a boolean and had multiple possible values

In which case `isEnabled` would be a shortcut like `return $this->state === ENABLED_STATE;` But for now it's only a simple boolean so I'd keep on isEnabled this way if we need to introduce a more advanced `getState` we won't have to integrate a BC break to change its type

Reply:
So in that spirit I would even modify the schema to use a `enabled` or `is_enabled` boolean in the DB

Reply:
If you absolutely want to use a  state because you think it would evolve in the future then replace it with a true integer or string and two possible values that are stored in const

Even if there are only two for now it will allow it to evolve later

Reply:
The whole point of [encapsulation](https://en.wikipedia.org/wiki/Encapsulation_(computer_programming)) is to hide the values or state.

From a developer point of view using the `FeatureFlag` class, it makes sense to have 2 different functions `getState()` and `isEnabled()` as both can have a different meaning for him.

As I said before, when you use `getState()` you want to extract some data (for example to pass it to another class like I do, to build something else, maybe return a structure like a JSON?).

When you use `isEnabled()` you dont care about the data of the FeatureFlag. You want to know "is it enabled? yes? no?" . You just want this information 😄  .

The flag has a state, it's a boolean. It's a state that can have 2 values 😄  . A boolean is like an enum whose possible values are `false` and `true`.

Both functions make sense to me, I really think the 2 functions can be kept. I personally prefer `getState()`, @jolelievre preferred `isEnabled()`. So both of us had a need 😄  it does not mean it has to be him or me. And we write code not only for us but also for all PrestaShop developers and we need to think about developer experience.

Having 2 functions with same implementation but different naming is not unusual when you use natural language. I dont see the problem with doing what we can to make the code more fluent, easier to use for developers.

Reply:
then maybe we could keep the two methods and add constants for the method `getState()` ? So that new constants can be defined later if new states are added, without doing a BC

Reply:
Mmm, I totally disagree with duplicating this `getState` with `isEnabled` But I think we misunderstood each other and you are overthinking this conversation 😅

The point of my initial feedback was not to add an alias of your existing function! It's just that the notion of `state` implies *multiple* states not just a boolean status That's why I suggested to rename into enabled which is more adapted

So I though either use a multi value integer state
Or use a boolean enabled|activated field

But not both ^^ And if you plan on making this state evolve I still think an integer or string would be more adapted, but if you prefer to keep a boolean state I'm ok with it then isEnabled becomes useless

tldr; If state becomes an int it's worth keep isEnabled() if not you can remove it

Reply:
A feature flag is either enabled or disabled, there's no in between. Avoid adding things "just in case",  you'll end up overengineering it. Think [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it).

Choose `isEnabled` or `getState`. Don't keep both.

Reply:
I really think it is valuable to be able to, in your code, write
```
$flag = $this->getFlagByName('my_flag');

if ($flag->isEnabled()) {
   $doA();
   $newObject = $this->create($a, $b, $flag->getState());
}
```
because it makes it easier to read as the function name describes the intent, not the implementation. It's [syntaxic sugar](https://en.wikipedia.org/wiki/Syntactic_sugar), and it aims to make the code easier to grasp for developers rather than for the implementation optimization. I think it's the same idea behind the `assertFalse()` and `assertTrue()` of PHPUnit (it could be replaced by `assertBool($bool)` or `assertIsBool($value)` ).

But I'm not willing to defend a position forever 😉  I will remove `getState();` following the feedback

Reply:
Then if you think isEnabled is simpler to understand that getState you can replace the whole `state` by `enabled` including in the DB no?

Reply:
In which case it's syntaxic sugar any more, it's just the field name 🤔

Reply:
@jolelievre Since the `state` db field name was chosen by @eternoendless in https://github.com/PrestaShop/PrestaShop/issues/23235 I am afraid of pushing another change that might be discussed by many people and just block this PR 😅  . This DB field could be named `state`, `status`, `enabled`, `is_enabled` and I dont want to ignite any more long debates, especially when I consider that this PR must be merged before the 15/04 ... and there is a few more that I need to also merge before the 15/04 and that are blocked by this one.

If however I am able to submit and merge all the PRs needed for the Feature Flag before 15/04 I will be happy to open a PR to update the name of the field, but I'm quite afraid that if I ask 3 person what the field name should be I will be told 3 different answers 😅  .

Reply:
My amazing invention is gone in https://github.com/PrestaShop/PrestaShop/pull/23703/commits/407f37ea8cf7de3d576608203da2b21ffa4b7be7 😢